presto-main/src/main/java/com/facebook/presto/sql/analyzer/MaterializedViewPlanValidator.java

```diff
@@ -32,15 +41,36 @@
 {
     private final Node query;
 
+    private final boolean analyzeColumnMapping;
+    private final Optional<Analysis> analysis;
+
+    private ImmutableList.Builder<List<TableColumn>> linkedBaseColumns;
+
     public MaterializedViewPlanValidator(Node query)
     {
+        this(query, true, Optional.empty());
+    }
+
+    public MaterializedViewPlanValidator(Node query, boolean analyzeColumnMapping, Optional<Analysis> analysis)
```

Comment:
This check needs to be fixed, we need analysis if column mapping needs to be analyzed.

Reply:
Exactly, I'm adding additional test cases now, and the tests capture this bug. Hold on a bit for review :) Don't want to waste your time.